climpred/utils.py

```diff
@@ -171,3 +172,16 @@ def copy_coords_from_to(xro_from, xro_to):
             f'xr.Dataset, found {type(xro_from)} {type(xro_to)}.',
         )
     return xro_to
+
+
+def _ensure_loaded(res):
```

Comment:
is this a good name? maybe `ensure_computed`

Reply:
also unsure about our conventions regarding `_` or no `_` for refactored functions

Reply:
I think just `_load_into_memory()` would work. But I also don't think this function is necessary at all since we shouldn't load things into memory for users. It should be returned as a lazy object to be computed by the user.

Reply:
Also `_` should be used if the user shouldn't be calling this directly from climpred. I.e. they shouldn't be doing `from climpred.utils import ensure_loaded` so it gets an underscore.

Reply:
> I think just `_load_into_memory()` would work. But I also don't think this function is necessary at all since we shouldn't load things into memory for users. It should be returned as a lazy object to be computed by the user.

This is the key point in this PR. We parallelise up to here because these lines need all computations do be done. 

I see the point of never calling compute. With my new quantile function we could also do this. Before xr quantile required loaded data.

Reply:
See comment in other thread. `dask` can handle setting up a graph in order. So right now this PR encourages chunking in space. So each spatial chunk gets a graph of things to do:

e.g. (not necessarily actually correct)

for i in N iterations:
1. shuffle
2. compute skill
3. compute persistence
4. bootstrap hist. 

After above is complete, compute quantiles.

Right now it is still doing this for every iteration. We'll get speed up on large arrays since we are chunking in space but won't get speed up from parallelizing the iterations.

Reply:
> Right now it is still doing this for every iteration. We'll get speed up on large arrays since we are chunking in space but won't get speed up from parallelizing the iterations.

I see this differently. right now for 5 bootstrapping, it creates the dask graph for 1.-4. 5 times. then it hits compute. and yes this compute might not be necessary.